azkaban-common/src/main/java/azkaban/server/HttpRequestUtils.java

```diff
@@ -233,8 +233,7 @@ public static void validatePreprocessFlowParameters(ExecutionOptions options, Pr
       }
     }
     // doesn't contain FlowParameters.FLOW_PARAM_MAX_RETRIES
-    else if (!options.isExecutionRetried()
-        && flowParameters.containsKey(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS)) {
+    else if (flowParameters.containsKey(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS)) {
```

Comment:
If the user didn't set `FLOW_PARAM_MAX_RETRIES` why should Azkaban set it on behalf of the user?

Reply:
I think it's a safe net, if ppl set `flow.retry.statuses` then they surely want to retry, but if they forget the `flow.max.retries` maybe should default to 1.

Also it's for backward compatibility, when ppl set `allow.restart.on.execution.stopped`, we default to have 1 retry, so the behavior should continue

Reply:
I see, thanks for clarifying! If maintaining backward compatibility then I think we should cover it inside the if of lines: 184-194. What do you think?

Reply:
Actually you are right, should not alter user input ExecutionOptions. Removing.